src/mvtseries.jl

```diff
@@ -916,3 +922,89 @@ function Base.mapslices(f, A::MVTSeries; dims)
     return res
 end
 
+########
+
+
+"""
+    rename_columns!(A::MVTSeries, new_names)
+    rename_columns!(A::MVTSeries, names_map)
+    rename_columns!(f::Function, A::MVTSeries)
+    rename_columns!(A::MVTSeries; prefix, suffix, replace)
+
+Rename the columns of an `MVTSeries` instance in-place. 
+* The new names can be specified in a list `new_names` (e.g., `Vector{Symbol}`
+  or `NTuple{N,Symbol}`), which must be of the exact length.
+* The new names can be given as a map (e.g., a dictionary or a named tuple),
+  where the old names are the keys and the new names are the values of
+  `names_map`. This makes it easy to rename only a few columns.
+* The new names can also be determined by a function `f(::Symbol) -> Symbol`
+  that transforms the old names into new names.
+* Some typical cases can be done using the keyword parameters `prefix`, `suffix`
+  and `replace`. In case `replace` is given together with one or both of
+  `suffix` and `prefix`, the replacement is applied first on the original name
+  and any suffix or prefix are applied after.
+"""
+function rename_columns! end
+export rename_columns!
+
+function rename_columns!(A::MVTSeries, newcolnames::_MVTSAxes2)
+    if size(A, 2) != length(newcolnames)
+        throw(ArgumentError("Incompatible number of new column names. Expected $(size(A,2)), got $(length(newcolnames))"))
+    end
+    # We make a new dictionary for the A.columns field. 
+    # We keep the existing values (TSeries containing views into A.values), but assign them to new keys (the new column names)
+    columns = getfield(A, :columns)
+    new_columns = typeof(columns)()
+    for (new_name, series) = zip(newcolnames, values(columns))
+        push!(new_columns, new_name => series)
+    end
+    setfield!(A, :columns, new_columns)
+    return A
+end
+
+function rename_columns!(f::Function, A::MVTSeries)
+    columns = getfield(A, :columns)
+    new_columns = typeof(columns)()
+    for (old_name, series) = pairs(columns)
+        new_name = f(old_name)
+        push!(new_columns, new_name => series)
+    end
+    setfield!(A, :columns, new_columns)
+    return A
+end
+
+function rename_columns!(A::MVTSeries, names_map::Union{NamedTuple,AbstractDict})
+    columns = getfield(A, :columns)
+    new_columns = typeof(columns)()
+    for (old_name, series) = pairs(columns)
+        new_name = get(names_map, old_name, old_name)
+        push!(new_columns, new_name => series)
+    end
+    setfield!(A, :columns, new_columns)
+    return A
+end
+
+rename_columns!(A::MVTSeries; replace=nothing, prefix=nothing, suffix=nothing) = rename_columns!(_rename_columns_func(replace, prefix, suffix), A)
```

Comment:
A vector of `new_names` would be better than a `names_map` since the construction and lookup times of vectors are cheaper than dictionaries (that's what I expect, I haven't actually measured it). Constructing a function would be best because there are no look up times and the function is pre-compiled, so the construction cost is fixed.

But in either of the three cases you can't escape the complexity of the logic here. You have 3 parameters, each of which can be `nothing`, but not all at the same time. So you have to handle all 7+1 cases. In the case of vector or map you can write a nested if-statement, but it wouldn't be any less complicated. In the case of a function, which is this implementation, you can use multiple dispatch instead of if-statements, which is the julian way.